### PR TITLE
Ld/upgrade to java 21

### PR DESCRIPTION
**Pull request by lindseydew**
## What does this change?

This upgrade the version of java that the lambdas run on to Java 21 as per our [departmental recommendations](https://github.com/guardian/recommendations/blob/257842468e9b8cd68ee967e568729eecafdddb46/scala.md?plain=1#L9)

## How to test

This has been regression tested on CODE

 

- [x] Able to read articles
- [x] Able to add articles
- [x] Able to remove articles
- [x] Able to delete a user record

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -67,7 +67,6 @@ val commonSettings: immutable.Seq[Def.Setting[_]] = List(
     "-deprecation",
     "-encoding",
     "UTF-8",
```
[the remark this answers is not in the document]
Comment (waisingyiu) [reply build.sbt:69]: I tried on my local machine and it compiled successfully with `-release:21` option.  Did I miss anything here?  (my local Java is `21.0.2-amzn`)

[the remark this answers is not in the document]
Comment (lindseydew) [reply build.sbt:69]: Ah my mistake, yes seems to be working fine for me too now. Perhaps I had been running against an earlier version of Java when I tested it before

[the remark this answers is not in the document]
Comment (aracho1) [reply build.sbt:69]: I also got this error initially and can confirm it went away once I switched to java 21

**Review by aracho1 — APPROVED** [review]
Tested with java 21 locally and looks good to me 👍

File: build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (rtyley) [reply build.sbt:69]: Yeah, the docs at https://docs.scala-lang.org/overviews/compiler-options/ are unhelpful there - I've opened https://github.com/scala/docs.scala-lang/pull/3009 to try to improve them!

**Review by rtyley — APPROVED** [review]
Awesome stuff - I haven't tested it out but very good to see updates to Java 21! ✨

File: .github/workflows/ci-mobile-save-for-later-user-deletion.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
       - name: Set up JDK
         uses: actions/setup-java@v3
```
[the remark this answers is not in the document]
Comment (lindseydew) [reply .github/workflows/ci-mobile-save-for-later-user-deletion.yaml:17]: Ah great thank you, I'll make a follow up PR for this.

File: build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (lindseydew) [reply build.sbt:69]: > Yeah, the docs at https://docs.scala-lang.org/overviews/compiler-options/ are unhelpful there - I've opened [scala/docs.scala-lang#3009](https://github.com/scala/docs.scala-lang/pull/3009) to try to improve them!

💖

**Pull request merged**